Finished project

kilozdazolik opened the pull request:
I Finished the coding tracker project I hope it's okay based on DRY and SOC principles.
I only finished 2 challenges because I want to move on on the learning with a different project.

kilozdazolik committed:
create db connection, main menu choices

kilozdazolik committed:
feat: implement session management features

This commit introduces a complete suite of features for managing sessions, including a new controller, model, and service layer.

- **Controller:** Added endpoints to handle creating, retrieving, and deleting sessions.
- **Service:** Implemented the business logic for session operations.
- **Model:** Defined the data structure for a session.

This change also includes a new view for displaying all sessions and a separate view for adding a new session.

kilozdazolik committed:
feat: add session management and stopwatch functions

- Added `update` and `delete` session in the controller and service.
- Implemented a stopwatch function.
- Created a new helper class.
- Expanded the UI to support the new features.

kilozdazolik committed:
feat: Implement filter and validation logic for sessions

This commit introduces new features for session management, including:
- A new filter method to retrieve sessions based on date.
- A refactored validation method to ensure end dates are not before start dates, following the DRY principle.

The service layer has been updated to handle these changes, improving code clarity and maintainability.

kilozdazolik committed:
Create README.md

kilozdazolik committed:
Update README.md

kilozdazolik committed:
fixed codacy issue

kilozdazolik committed:
Merge remote-tracking branch 'origin/main'

andymartinez1 commented:
Hi @kilozdazolik! It seems you are missing your .sln and .csproj files. Can you please include those in your solution?

kilozdazolik committed:
feat: upload .csproj file

kilozdazolik commented:
Hi @andymartinez1! I have included my .csproj file in the project.

andymartinez1 reviewed (changes requested):
Hi @kilozdazolik. Great job on the project, especially doing one of the challenges. 

I have noticed a few small issues that will need to be fixed before I can approve it. 

1. The date format is wrong when adding a new session:
<img width="1086" height="333" alt="image" src="https://github.com/user-attachments/assets/b69eea30-c473-4da8-94b0-7b229452a7b2" />
It only works when I enter mm-dd-yyyy instead of dd-mm-yyyy
<hr/>
2. When editing a session, it uses a different date format completely:
<img width="1023" height="248" alt="image" src="https://github.com/user-attachments/assets/51cbfdf9-e66d-4572-99f0-c00719ce340e" />
You should keep this consistent with one format
<hr/>
3. When viewing all sessions, there seems to be a bug. Viewing sessions multiple times shows the list multiple times. Here's what I mean:
<img width="768" height="1091" alt="image" src="https://github.com/user-attachments/assets/f13b998b-60b5-41de-963f-3ddf22b2cc36" />
It keeps adding the list over and over. It should only display once.
<hr/>
Please fix these issues so I can approve it. If you need any help or have any questions, please feel free to reach out!

kilozdazolik committed:
feat: Add new methods and fix date issues

This commit addresses several issues and introduces new functionality.

- **Fix:** Resolved date validation issues.
- **Feat:** Implemented a new `showWithPause` method to clear the console.
- **Feat:** Added a `helper` method for date validation.
- **Refactor:** Made minor UI adjustments for a better user experience.

kilozdazolik committed:
fixed codacy issue

kilozdazolik commented:
Hi **@andymartinez1** ! I have fixed the issues that you have mentioned.

andymartinez1 reviewed (approved):
Hi @kilozdazolik. Great work fixing all of the issues! It's now working flawlessly. Project approved
